### PR TITLE
Move UI visualizer code out of this repo

### LINKED ISSUE #1129: Move flowsheet visualizer UI code to new repositories
The flowsheet visualizer (UI) code doesn't need to be in the PSE core.
It should be broken out into its own repository, *e.g.*, **idaes-ui**.
The UI code will be a Python repository that is pip installable (`pip install idaes-ui`).
Inside that repository will be the Python code and the JavaScript application(s) -- which for now is the flowsheet visualizer.
While we are at it, this is a good time to break the core JS diagramming code of the flowsheet visualizer into its own repository that can be installed as a node package (using npm), *e.g.*, **idaes-fsvis-js**.

In summary:

* `idaes-ui` will have Python code for the flowsheet visualization server, and Javascript code for the flowsheet visualization web UI, and is installable from pip as a Python package
* `idaes-fsvis-js` will have the JS code that implements the flowsheet visualization diagram, and is installable from npm as a JS package
* None of the above code, except perhaps a small amount of interface code to invoke the UI, will be in idaes-pse any longer. The UI can be installed as a separate Python package, which could also be made an optional package for idaes-pse like `pip install idaes-pse[ui]` (these details can be decided later).

Note: assigning a few people to this issue to allow for discussion. I expect to do most or all the code movement myself (Dan).

### PR DESCRIPTION
**Pull request by dangunter**
Fixes #1129

Related to https://github.com/IDAES/idaes-ui/pull/11

## Changes proposed in this PR:
- Remove visualizer code
- Add tests to verify correct operation
- Update docs accordingly

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by dangunter** [comment]
@andrewlee94 w.r.t. installation instructions I agree, and it was on my to-do list. In a perfect world that would go first. But I have noticed that the current version in main has a bug in the stream table, that didn't seem to lead to any complaints (but looks bad). So in the interest of simplicity, I'd like to just move forward with this PR and then add back instructions once I've got a less borked version of the UI working in the new repo.

**Review by lbianchi-lbl — CHANGES_REQUESTED** [review]
I've left a suggestion to consider whether we need the added complexity of the wrapper class for this particular case since there's only one call site to handle. Otherwise looks good.

File: idaes/core/base/flowsheet_model.py [hunk 1]
```diff
@@ -229,15 +269,15 @@ def visualize(self, model_name, **kwargs):
...
+            model_name : The name of the model
+
         Keyword Args:
             **kwargs: Additional keywords for :func:`idaes.core.ui.fsvis.visualize()`
 
         Returns:
             None
         """
-        visualize(self, model_name, **kwargs)
+        ui.visualize(self, model_name, **kwargs)
```
[the remark this answers is not in the document]
Comment (dangunter) [reply idaes/core/base/flowsheet_model.py]: I wanted to allow for other functions, e.g. for diagnostics

**Comment by CopyDemon** [comment]
I agree with the changes. I think it's a good idea to remove the UI from the main code and add it to its own repo. This will make the code more modular and easier to maintain.

I also agree with Andrew's comment that we should add some sort of pointer to documentation on how to install and set up the UI in the main IDAES docs. This will make it easier for users to get started with the UI. Or we can make some automation install when they install IDAES.

**Review by CopyDemon — APPROVED** [review]

**Comment by lbianchi-lbl** [comment]
- The conflict should be trivial (files being deleted here were modified in #1133)
- We should be able to merge this soon

**Comment by lbianchi-lbl** [comment]
Waiting on @dangunter and/or @lbianchi-lbl to fix the git conflicts.

**Review by lbianchi-lbl — APPROVED** [review]

**Pull request merged**